### PR TITLE
Prompt user to logout before connecting to WPCom site

### LINKED ISSUE #5166: Signin: Prompt if signing in to wpcom via self-hosted form, if already a wpcom acct in the app.
The self-hosted signin form will allow a user to enter credentials and URL to a wpcom site, and authenticate accordingly.  This is fine if there is not currently a default wpcom account in the app but could be problematic otherwise. 

Lets add a check for an existing default wpcom account and prompt the user they should first sign out of it.

### PR DESCRIPTION
**Pull request by yaelirub**
Fixes #5166 

When a user chooses to add a self hosted site but inputs a WPCom site address, we check to see is a default WPCom account already exists in the app
and if it does we prompt the user to logout before connecting to the other site.

selfhostedprompt

To test:
1. Start by being signed in to a WordPress.com account.
2. Go to the My Sites tab and view the list of sites.
3. Tap the + button in the nav bar and choose "Add self-hosted site"
4. On the site address screen, enter the URL for a WordPress.com site that does not belong to the WordPress.com account that's currently logged into the app.
5. On the next screen, enter the WordPress.com credentials for the account that owns the site at the URL entered on the previous screen.
6. Repeat steps 1-5 with a site that is not WPCom.
7. See that the alert does not show and you are pushed to the credentials screen.

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.
* Added alert to prompt the user to logout before connecting to a wordpress.com site, if they are already logged in with a wordpress.com account

### CODE REVIEW

**Comment by michelleweber** [comment]
In the pop-up, "log out" should be two words and you need a period at the end of the sentence.

**Comment by yaelirub** [comment]
@frosty , I agree. It would be a better user experience to allow the user to logout from the alert.

File: Podfile [hunk 1]
```diff
@@ -135,7 +135,7 @@ target 'WordPress' do
 
     pod 'WordPressAuthenticator', '~> 1.1.8-beta.1'
     #pod 'WordPressAuthenticator', :path => '../WordPressAuthenticator-iOS'
-    #pod 'WordPressAuthenticator', :git => 'https://github.com/wordpress-mobile/WordPressAuthenticator-iOS.git' , :commit => '82900447393674532003833cc1f3a4ad1c53997f'
+    #pod 'WordPressAuthenticator', :git => 'https://github.com/wordpress-mobile/WordPressAuthenticator-iOS.git' , :commit => '895563cdd426a1b6e72702cfa40f9595ea470b74'
```
Comment (jtreanor) [inline Podfile]: @yaelirub  It looks like your problem may be here. You changed the commit hash but this line is actually commented out.

I think the change you actually want to make is changing above to `pod 'WordPressAuthenticator', '~> 1.1.8-beta.2'`. This should fix the conflict with lottie

Comment (yaelirub) [reply Podfile]: Hey @jtreanor  when I update to: pod 'WordPressAuthenticator', '~> 1.1.8-beta.2' I get:
screen shot 2019-01-24 at 8 09 17 am

I feel like Im in CocoaPods hell and not sure what's going on :(

Comment (jklausa) [reply Podfile]: @yaelirub I _think_ if you do `pod repo update` it should be fine

Comment (jtreanor) [reply Podfile]: Unfortunately not in this case 😞 Using the new WordPressAuthenticator depends on this PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/10827. There were lots of updates to the dependency graph so thats why we are seeing the error.

**Comment by ctarda** [comment]
I pushed a [commit](https://github.com/wordpress-mobile/WordPress-iOS/pull/10862/commits/c37318fbe4ed160c4386fd5bcf332001555d3495)  to integrate some breaking changes introduced when releasing WordPressKit 1.8.0-beta.1

The diff in this commit is relevant to the development of the SiteCreation feature, and has already been reviewed and approved in [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/10877) #10877 

I will add comments marking the specific changes introduced by my commit

File: WordPress/Classes/Models/SiteSegment.swift [hunk 2]
```diff
@@ -1,44 +0,0 @@
-
```
Comment (ctarda) [inline WordPress/Classes/Models/SiteSegment.swift]: Change introduced to integrate WordPressKit 1.8.0-beta.1

File: WordPress/Classes/ViewRelated/Site Creation/SiteSegments/SiteSegmentsCell.swift [hunk 3]
```diff
@@ -1,6 +1,20 @@
 import UIKit
 import Alamofire
 import Gridicons
+import WordPressKit
```
Comment (ctarda) [inline WordPress/Classes/ViewRelated/Site Creation/SiteSegments/SiteSegmentsCell.swift]: Change introduced to integrate WordPressKit 1.8.0-beta.1

File: WordPress/WordPressTest/SiteCreation/SiteAssemblyServiceTests.swift [hunk 4]
```diff
@@ -14,7 +14,8 @@ class SiteAssemblyServiceTests: XCTestCase {
             title: "A title",
             subtitle: "A subtitle",
             icon: URL(string: "https://s.w.org/style/images/about/WordPress-logotype-standard.png")!,
-            iconColor: .red)
+            iconColor: "#FF0000",
```
Comment (ctarda) [inline WordPress/WordPressTest/SiteCreation/SiteAssemblyServiceTests.swift]: Change introduced to integrate WordPressKit 1.8.0-beta.1

File: WordPress/WordPressTest/SiteCreation/SiteCreatorTests.swift [hunk 5]
```diff
@@ -18,7 +18,8 @@ class SiteCreatorTests: XCTestCase {
             title: "A title",
             subtitle: "A subtitle",
             icon: URL(string: "https://s.w.org/style/images/about/WordPress-logotype-standard.png")!,
-            iconColor: .red)
+            iconColor: "#FF0000",
```
Comment (ctarda) [inline WordPress/WordPressTest/SiteCreation/SiteCreatorTests.swift]: Change introduced to integrate WordPressKit 1.8.0-beta.1

File: WordPress/WordPressTest/SiteCreation/SiteSegmentTests.swift [hunk 6]
```diff
@@ -44,12 +44,8 @@ final class SiteSegmentTests: XCTestCase {
         XCTAssertEqual(subject?.icon, MockValues.icon)
     }
 
-    func testIconColorIsNotMutated() {
-        XCTAssertEqual(subject?.iconColor, MockValues.iconColor)
-    }
-
     func testSiteTypesWithSameIdAreEqual() {
-        let secondSiteType = SiteSegment(identifier: MockValues.identifier, title: "Another title", subtitle: "It does not matter", icon: MockValues.icon, iconColor: .white)
+        let secondSiteType = SiteSegment(identifier: MockValues.identifier, title: "Another title", subtitle: "It does not matter", icon: MockValues.icon, iconColor: "#FF0000", mobile: true)
```
Comment (ctarda) [inline WordPress/WordPressTest/SiteCreation/SiteSegmentTests.swift]: Change introduced to integrate WordPressKit 1.8.0-beta.1

File: WordPress/WordPressTest/Test Data/site-segment.json [hunk 7]
```diff
@@ -3,5 +3,6 @@
     "segment_type_subtitle": "Publish a collection of posts.",
     "icon_URL": "https://wordpress.com/icon/blogger.png",
     "icon_color": "FFFAFA",
-    "segment_id": 1
```
Comment (ctarda) [inline WordPress/WordPressTest/Test Data/site-segment.json]: Change introduced to integrate WordPressKit 1.8.0-beta.1

**Pull request merged**